### PR TITLE
PSR-251 Adding the partnership UTR following the IdentityType approach

### PR DESCRIPTION
**Pull request by AlessandroCinque**
https://jira.tools.tax.service.gov.uk/browse/PSR-251 Adding the partnership UTR

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -369,14 +369,14 @@ GET         /:srn/submit-file-upload-too-many-rows
...
-GET         /:srn/change-partnership-recipient-utr                                      controllers.nonsipp.common.PartnershipRecipientUtrController.onPageLoad(srn: Srn, index: Max5000 = 1, mode: Mode = CheckMode)
-GET         /:srn/change-partnership-recipient-utr/:index                               controllers.nonsipp.common.PartnershipRecipientUtrController.onPageLoad(srn: Srn, index: Max5000, mode: Mode = CheckMode)
-POST        /:srn/change-partnership-recipient-utr                                      controllers.nonsipp.common.PartnershipRecipientUtrController.onSubmit(srn: Srn, index: Max5000 = 1, mode: Mode = CheckMode)
-POST        /:srn/change-partnership-recipient-utr/:index                               controllers.nonsipp.common.PartnershipRecipientUtrController.onSubmit(srn: Srn, index: Max5000, mode: Mode = CheckMode)
+GET         /:srn/partnership-recipient-utr:subject                                             controllers.nonsipp.common.PartnershipRecipientUtrController.onPageLoad(srn: Srn, index: Max5000 = 1, mode: Mode = NormalMode, subject: IdentitySubject)
```
Comment (kberec) [inline conf/app.routes:372]: @AlessandroCinque What do you think about `utr-of-:subject` instead?

File: conf/app.routes [hunk 2]
```diff
@@ -423,12 +423,12 @@ GET         /:srn/change-amount-of-loan-payments/:index
...
-POST        /:srn/submit-check-answers-loan                                             controllers.nonsipp.loansmadeoroutstanding.LoansCYAController.onSubmit(srn: Srn, checkOrChange: CheckOrChange = Check)
-GET         /:srn/change-check-answers-loan                                             controllers.nonsipp.loansmadeoroutstanding.LoansCYAController.onPageLoad(srn: Srn, index: Max5000 = 1, checkOrChange: CheckOrChange = Change)
-GET         /:srn/change-check-answers-loan/:index                                      controllers.nonsipp.loansmadeoroutstanding.LoansCYAController.onPageLoad(srn: Srn, index: Max5000, checkOrChange: CheckOrChange = Change)
-POST        /:srn/change-submit-check-answers-loan                                      controllers.nonsipp.loansmadeoroutstanding.LoansCYAController.onSubmit(srn: Srn, checkOrChange: CheckOrChange = Change)
+GET         /:srn/check-answers-loan:subject                                                controllers.nonsipp.loansmadeoroutstanding.LoansCYAController.onPageLoad(srn: Srn, index: Max5000 = 1, checkOrChange: CheckOrChange = Check, subject: IdentitySubject)
```
Comment (kberec) [inline conf/app.routes:426]: Needs reverting LoansCYAController doesn't need subject argument. It's always loan recipient.

File: app/controllers/nonsipp/loansmadeoroutstanding/LoansCYAController.scala [hunk 3]
```diff
@@ -52,7 +52,8 @@ class LoansCYAController @Inject()(
   def onPageLoad(
     srn: Srn,
     index: Max5000,
-    checkOrChange: CheckOrChange
+    checkOrChange: CheckOrChange,
+    subject: IdentitySubject
```
Comment (kberec) [inline app/controllers/nonsipp/loansmadeoroutstanding/LoansCYAController.scala:56]: Needs reverting, as this doesn't need subject argument. It's always loan recipient.

File: app/navigation/nonsipp/LoansMadeOrOutstandingNavigator.scala [hunk 4]
```diff
@@ -164,69 +164,69 @@ object LoansMadeOrOutstandingNavigator extends JourneyNavigator {
             .onPageLoad(srn, index, NormalMode)
         case Some(_) =>
           controllers.nonsipp.loansmadeoroutstanding.routes.LoansCYAController
-            .onPageLoad(srn, index, Check)
+            .onPageLoad(srn, index, Check, IdentitySubject.LoanRecipient)
```
Comment (kberec) [inline app/navigation/nonsipp/LoansMadeOrOutstandingNavigator.scala:167]: Same for these below.

File: conf/app.routes (hunk 1 above)
Comment (AlessandroCinque) [reply conf/app.routes:372]: Yeah that makes more sense indeed, I will change it in the next commit

File: conf/app.routes (hunk 2 above)
Comment (AlessandroCinque) [reply conf/app.routes]: Indeed I wasn't extra sure of that change, I just reverted it

File: conf/app.routes [hunk 5]
```diff
@@ -369,14 +369,14 @@ GET         /:srn/submit-file-upload-too-many-rows
...
-GET         /:srn/change-partnership-recipient-utr                                      controllers.nonsipp.common.PartnershipRecipientUtrController.onPageLoad(srn: Srn, index: Max5000 = 1, mode: Mode = CheckMode)
-GET         /:srn/change-partnership-recipient-utr/:index                               controllers.nonsipp.common.PartnershipRecipientUtrController.onPageLoad(srn: Srn, index: Max5000, mode: Mode = CheckMode)
-POST        /:srn/change-partnership-recipient-utr                                      controllers.nonsipp.common.PartnershipRecipientUtrController.onSubmit(srn: Srn, index: Max5000 = 1, mode: Mode = CheckMode)
-POST        /:srn/change-partnership-recipient-utr/:index                               controllers.nonsipp.common.PartnershipRecipientUtrController.onSubmit(srn: Srn, index: Max5000, mode: Mode = CheckMode)
+GET         /:srn/partnership-recipient-utr-:subject                                             controllers.nonsipp.common.PartnershipRecipientUtrController.onPageLoad(srn: Srn, index: Max5000 = 1, mode: Mode = NormalMode, subject: IdentitySubject)
```
Comment (kberec) [inline conf/app.routes:372]: Subject can be long. And it is not necessary a recipient. So I'd shorten the first part. E.g. `/utr-of-:subject`

**Review by kberec — DISMISSED** [review]

**Review by kberec — DISMISSED** [review]

**Pull request merged**